pkg/chunk/series_store.go

```diff
@@ -131,6 +132,15 @@ func (c *seriesStore) Get(ctx context.Context, userID string, from, through mode
 		return nil, err
 	}
 
+	// inject artificial __cortex_shard__ labels if present in the query. GetChunkRefs guarantees any chunk refs match the shard.
+	shard, _, err := astmapper.ShardFromMatchers(allMatchers)
```

Comment:
You might want to do that in `GetChunkRefs` in case someone (Loki for instance does that) uses the refs first then fetches chunks after.

Reply:
Well might not work because `fetcher.FetchChunks` might validate the metric. To be verified.

Reply:
This is injecting a `cortex_shard` label into each of the resulting series because mapped queries expect its existence.

If we query `metric{__cortex_shard__="1_of_2"}`, we expect the responses from the querier to include this `__cortex_shard__="1_of_2"` label.

`GetChunkRefs` does only return the shard-specific series if a shard label is present in the query. This logic is handled below in `lookupSeriesByMetricNameMatcher{s,}`